### PR TITLE
Feature/select role

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Create a portal module that allows a certain role to select another role for his device (like a technician registering devices with the portal)

# Impacts
Portal

# Delete branch after merge
YES

# NEWS file entries
## Enhancements
* Added portal module for selecting a role for the device being registered on the portal

### CODE REVIEW

**Comment by fdurand** [comment]
Applied on a setup and it works.
Not against merging for 7.3

File: html/captive-portal/lib/captiveportal/DynamicRouting/Module/SelectRole.pm [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+BEGIN { extends 'captiveportal::PacketFence::DynamicRouting::Module::SelectRole'; }
+
+=head1 NAME
+
+captiveportal::DynamicRouting::Module::SelectRole - SelectRole Controller for captiveportal
+
+=head1 DESCRIPTION
+
+[enter your description here]
```
Comment (fdurand) [inline html/captive-portal/lib/captiveportal/DynamicRouting/Module/SelectRole.pm:12]: Module to select a new role for the device being registered

Comment (julsemaan) [reply html/captive-portal/lib/captiveportal/DynamicRouting/Module/SelectRole.pm:12]: Given its the «extension» point, we usually don't put a description there
At least, thats how the others are, IDK if it was the way it was meant to be but I find it to be a good idea

**Comment by dwlfrth** [comment]
Push to 7.4 to allow documentation writeup

**Comment by dwlfrth** [comment]
@whitx test and check if you can get some documentation

**Comment by fdurand** [comment]
Agree to merge

**Comment by julsemaan** [comment]
@fdurand then you can hit that beautiful merge button.

Since @whitx has a commit in this branch the CLA check won't pass but I think that doesn't matter.
I guess @cgx can confirm

**Comment by julsemaan** [comment]
image

**Pull request merged**